packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/ObserverFlutterApiWrapper.java

```diff
@@ -0,0 +1,101 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.camerax;
+
+import android.util.Log;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import androidx.camera.core.CameraState;
+import androidx.camera.core.ZoomState;
+import androidx.lifecycle.Observer;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugins.camerax.GeneratedCameraXLibrary.ObserverFlutterApi;
+import java.util.Objects;
+
+/**
+ * Flutter API implementation for {@link Observer}.
+ *
+ * <p>This class may handle adding native instances that are attached to a Dart instance or passing
+ * arguments of callbacks methods to a Dart instance.
+ */
+public class ObserverFlutterApiWrapper {
+
+  private static final String TAG = "ObserverFlutterApi";
+
+  private final BinaryMessenger binaryMessenger;
+  private final InstanceManager instanceManager;
+  private ObserverFlutterApi observerFlutterApi;
+
+  @VisibleForTesting @Nullable public CameraStateFlutterApiWrapper cameraStateFlutterApiWrapper;
+  @VisibleForTesting @Nullable public ZoomStateFlutterApiImpl zoomStateFlutterApiImpl;
+
+  /**
+   * Constructs a {@link ObserverFlutterApiWrapper}.
+   *
+   * @param binaryMessenger used to communicate with Dart over asynchronous messages
+   * @param instanceManager maintains instances stored to communicate with attached Dart objects
+   */
+  public ObserverFlutterApiWrapper(
+      @NonNull BinaryMessenger binaryMessenger, @NonNull InstanceManager instanceManager) {
+    this.binaryMessenger = binaryMessenger;
+    this.instanceManager = instanceManager;
+    observerFlutterApi = new ObserverFlutterApi(binaryMessenger);
+  }
+
+  /**
+   * Sends a message to Dart to call {@link Observer.onChanged} on the Dart object representing
+   * {@code instance}.
+   */
+  public void onChanged(
+      @NonNull Observer<?> instance,
+      @NonNull Object value,
+      @NonNull ObserverFlutterApi.Reply<Void> callback) {
+
+    // Cast value to type of data that is being observed if supported by this plugin.
+    if (value instanceof CameraState) {
```

Comment:
It looks like the types need to be supported here an in `plugins/camerax/LiveDataHostApiImpl.java`. Can you define a list or some other object that is shared in the java files for what types are supported? If not can you add comments that link the implementation requirements for adding a new type?

Reply:
Also I would suggest refactoring these conversions into their own methods used in this class.

Reply:
I added a comment pointing to the instructions I wrote in `pigeons/camerax_library.dart`. I can't use the enum directly here because this is a callback from Android, but it is defined in `GeneratedCameraXLibrary.java`. I also mentioned it in the comment.